website/content/docs/rfcs/signed-commits.mdx

```diff
@@ -0,0 +1,100 @@
+## Summary
+This RFC proposes discontinuing the enforcement of signed commits via branch
+protection rules in our GitHub repositories. The current practice requires every
+commit to be signed, with the intention of ensuring authenticity and authorship
+verification. However, this enforcement introduces complexities and barriers
+that may outweigh its perceived security benefits. The primary goal of this
+proposal is to streamline the contribution process, making it more accessible to
+casual contributors while maintaining adequate security practices through other
+means. This document outlines the rationale, potential impacts, and alternative
+solutions to ensure the integrity and security of our codebase.
+
+## Problem Statement
+The requirement for signed commits is intended to enhance the security of our
+codebase by verifying the identity of contributors. However, this policy has
+introduced several challenges:
+
+* **Onboarding Difficulty**: New contributors often face a steep learning curve
+  to configure commit signing correctly.
+* **Increased Overhead**: Regular contributors spend additional time ensuring
+  that their commits are signed, which can slow down the development process.
+* **Exclusion of Contributors**: Contributors without the technical know-how or
+  access to necessary tools are discouraged from contributing, which reduces
+  community engagement and diversity.
+* **Temporal Validity of Verification**: The validity of commit signatures is
+  temporal. If a contributor deletes their SSH key from GitHub's UI, previously
+  "verified" commits may become "unverified," compromising the long-term
+  reliability of signature verification.
+* **Rebase Merge Issues**: Ideally, we would like to rebase pull requests when
+  merging to maintain the original commit history of the PR. However, due to a
+  long-standing bug in GitHub, all signatures in individual commits become
+  invalidated during a rebase, causing them to appear as unverified.
+* **Lack of Proper Trust Verification**: We currently have no means to verify
+  the Web of Trust of signed commits, which means we cannot reliably detect if a
+  contributor's GitHub account has been compromised.
+
+## User-facing description
+We are proposing to stop requiring signed commits for contributions to our
+GitHub repositories from casual contributors. This change aims to simplify the
+contribution process, making it easier for both new and existing contributors to
+participate. We will continue to require core Maintainers to sign their commits.
+
+## Technical Description
+To implement this change, we will:
+
+1. Update the branch protection rules in GitHub to remove the requirement for
+   signed commits.
+2. Add a GitHub Actions Workflow to ensure Maintainer commits are always
+   verified.
+2. Communicate the change to all contributors through appropriate channels
+   (e.g., mailing lists, GitHub discussions).
+3. Update our contribution guidelines to reflect the new policy.
```

Comment:
FWIW, our existing [CONTRIBUTING.md](https://github.com/openbao/openbao/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-dco-sign-off-policy) only mentions DCO and not commit GPG signing, so formalizing this policy actually brings us in line with the existing guidelines. :-)